src/conversions.cc

```diff
@@ -210,7 +210,16 @@ core::matrix_t floatSeqSeqToMatrix(const floatSeqSeq& input,
   matrix_t out(rows, cols);
   for (size_type r = 0; r < rows; ++r) {
     const floatSeq& row = input[(CORBA::ULong)r];
-    for (size_type c = 0; c < cols; ++c) out(r, c) = row[(CORBA::ULong)c];
+    size_type cols_in_row = (size_type)row.length();
+    if (cols_in_row != expectedCols)
```

Comment:
Did you try with `expectedCols = -1` which means that no expected size is provided. I would rather write
`if (cols_in_row != cols)`